### PR TITLE
InputOf update - authors, tests and ctors.

### PR DESCRIPTION
**Pull request by ixmanuel**

### CODE REVIEW

**Comment by ixmanuel** [comment]
@yegor256 here are the updates for InputOf in a new PR. Thanks.

File: src/test/java/org/cactoos/io/InputOfTest.java [hunk 1]
```diff
@@ -240,36 +240,42 @@ public void readsInputIntoBytes() throws IOException {
...
+        final String ends = "is now!";
         MatcherAssert.assertThat(
-            "Can't receive string buffer",
+            "Can't receive a string buffer",
             new String(
                 new InputAsBytes(
                     new InputOf(
-                        new StringBuffer("The future is now!")
+                        new StringBuffer(starts)
+                            .append(ends)
```
Comment (alex-semenyuk) [inline src/test/java/org/cactoos/io/InputOfTest.java]: @ixmanuel The same is above

File: src/test/java/org/cactoos/io/InputOfTest.java [hunk 2]
```diff
@@ -240,36 +240,42 @@ public void readsInputIntoBytes() throws IOException {
...
+        final String ends = "then it exists!";
         MatcherAssert.assertThat(
-            "Can't receive string builder",
+            "Can't receive a string builder",
             new String(
                 new InputAsBytes(
                     new InputOf(
-                        new StringBuilder("Name it, then it exists!")
+                        new StringBuilder(starts)
+                            .append(ends)
```
Comment (alex-semenyuk) [inline src/test/java/org/cactoos/io/InputOfTest.java]: @ixmanuel Let's use `new JoinedText`

File: src/main/java/org/cactoos/io/InputOf.java [hunk 3]
```diff
@@ -188,9 +184,21 @@ public InputOf(final StringBuilder builder, final Charset cset) {
...
+     *
+     * @param buffer The string's buffer
+     * @param cset The charset
+     */
+    public InputOf(final StringBuffer buffer, final Charset cset) {
         this(() -> {
             return new IoCheckedScalar<InputStream>(
-                () -> new ByteArrayInputStream(buffer.toString().getBytes())
+                () -> new ByteArrayInputStream(
+                    buffer.toString().getBytes(cset)
```
Comment (alex-semenyuk) [inline src/main/java/org/cactoos/io/InputOf.java]: @ixmanuel maybe better `new TextAsBytes(buffer.toString(), cset)`, but seems need additional constructor

Comment (ixmanuel) [inline src/main/java/org/cactoos/io/InputOf.java]: @alex-semenyuk seems good. Thanks.

File: src/test/java/org/cactoos/io/InputOfTest.java (hunk 2 above)
Comment (ixmanuel) [inline src/test/java/org/cactoos/io/InputOfTest.java]: Yes it's a good idea but here in the test, .append is just to illustrate the use of InputOf in context.

File: src/test/java/org/cactoos/io/InputOfTest.java (hunk 1 above)
Comment (ixmanuel) [inline src/test/java/org/cactoos/io/InputOfTest.java]: @alex-semenyuk same as above.

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**